### ISSUE TITLE
Add `ScriptsToInclude` to the Module Manifest.

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#24253]
### Summary of the new feature / enhancement

To extent on issue [`#18740` Best practice: import functions in a module`](https://github.com/PowerShell/PowerShell/issues/18740) with a more practical propose:
Adding a `ScriptsToInclude` to [Module manifest elements](https://learn.microsoft.com/powershell/scripting/developer/module/how-to-write-a-powershell-module-manifest#module-manifest-elements) would allow IMO allow for a safer (in terms of making module design misstakes) way to include script than dot-sourcing each script from the `<module>.psm1` file.

### Proposed technical implementation details (optional)

My expectation of a `ScriptsToInclude` element, is that it loads one or more **scripts** into the module **as a function** using the script name as the function name similar as installing a script (`Install-Script -Name <MyScript>`) from the PowerShell Gallery. 

Maintaining (as debugging the script apart (or not) from the module would also be easier as I might test it by using it's file name `.\MyScript.ps1` (without dot-sourcing first) and would be able compare the result with the loaded module function (just `MyScript`).

**Comment by rhubarb-geek-nz** [comment]
Thanks for the link to the previous issue. I thought that had [the best solution](https://github.com/PowerShell/PowerShell/issues/18740#issuecomment-1341183639).

The ideal form of a script module is a single psm1 file and no dot-sourced scripts. Your approach is trying to shift the responsibility of what the module packager should be doing onto the import process.

Regarding the testing, Pester will let you test complete modules so all the tools are already there.

**Comment by rhubarb-geek-nz** [comment]
> Besides by rebuilding and reloading the whole module, you will loose the reference of how the the orginal function reacted in the prior full module release..

Not really. That is all part of Test-Driven-Development and software versioning. 

You should always be able to refer to how a previous revision behaves by simply running the previous revision of the module.

Likewise your tests should be written first and define how you want the module to behave. You write the code to meet the tests. 

Your tests should alert you when your module changes in expected behaviour.

For every bug, defect, change etc add more tests to your suite so you are defining exactly how your module behaves and prevent regressions.

**Comment by kilasuit** [comment]
I can understand the ask, thou we already have `NestedModules` for importing into the Module Scope (which you could do what you want with) or the `ScriptsToProcess` field which will import any script you specify into the session before loading the module that's to be imported, which also could be used to do what you want, just differently.

Not all scripts to add are a single function, so adding this is not ideal either. 

I personally can't see a need for this especially as we recommend a build and test process that rolls up into a single psm1 when working with script modules. Due to the amount of performance gains you get with it

Regarding 

> Besides, by rebuilding and reloading the whole module, you will loose the reference of how the the orginal function reacted in the prior full module release..

One of the ways that i've used in developing locally in my own projects in a way that enables me to not loose this and until I am happy with how it functions, I may end up with 5..100 run sessions in that time split across devices and across multiple different days.
I do this with Transcription enabled, and if it's a simple small change to a function definiton I often will just overwrite it in the Function:\ path & test out how this is manually.
I then tend to run each new build of a module in a side-by-side session for inspection of intention vs actual behavior. 
That way I can inspect as I go and if I want to dump out to disk any session info I use Export-CLIXML for this so that I can inspect it at a later date if needs be.

**Comment by iRon7** [comment]
@kilasuit,

Thank you for you feedback, comments and suggestions.

Maybe this issue (and the one #18740 referring to) should be more of a discussion topic.

I guess that the point I am trying to make comes down to that PowerShell module (`.psm`) files aren't (in a broad sense) as safe to work with as they could be. Using the `NestedModules` or `ScriptsToProcess` fields won't change that.

You might compare this to a general PowerShell Data (`.psd1`) file: one could rename the `.psd1` file to a `.ps1` file and just invoke (dot-source) that, the results are the same but it is *not recommended because it is not safe*. PowerShell Data (`.psd1`) file are protected by the PowerShell **Restricted Language Mode**. (I bet that somewhere under the hood, the `.SafeGetValue()` method is used for this.)

For a PowerShell module (`.psm1`) files (or any other way to define the module), there are no boundaries, I can ***only* invoke** script (without any restrictions) similar to invoking a `.ps1` script in **Full Language Mode**, where essentially I only need to ***define*** a set of specific items as `functions`, `classes`, `enums` and statements as `requires` and `using` and possibly a few things more as (hardcoded) variable definitions (anything else could be done via the `ScriptsToProcess` field). You might almost think of a "**PowerShell *Definition* Language Mode**". But than, there is already a restricted manifest that is further bound by its predefined settings. Unfortunately each manifest settings is ***invoking*** one or more (*cascaded*) scripts to define the items (as functions, classes, etc.) rather than ***loading*** these *definitions* into the module...
